### ISSUE TITLE
php: Could we enable the imap extension on darwin by default?

### ISSUE THREAD

**Issue by ebauger** [NixOS/nixpkgs#190514]
### Describe the bug
this package ([php80Extensions.imap](https://search.nixos.org/packages?channel=22.05&show=php80Extensions.imap&from=0&size=50&sort=relevance&type=packages&query=php80Extensions.imap))  is not available for any platforms.
### Steps To Reproduce
When I try to install the package from the shell

### Expected behavior
To be available for popular platform like x86/amd64. arm64
### Notify maintainers
@aanderse 
@drupol : Answered me by email. Merci :)
@etu
@globin 
@Ma27 
@talyz 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
these 52 paths will be fetched (84.74 MiB download, 528.13 MiB unpacked):
  /nix/store/1m5r3xkfbgcnfrz99kkjdb5h29v36rp6-binutils-2.38
  /nix/store/21vk5rmy1a4p3kx22wzrdrllcmy8x5hl-libcxxabi-11.1.0
  /nix/store/2qv84lnl9m6spbfvqdp0a5zc20zg06nd-patch-2.7.6
  /nix/store/3azxqs52ssvb9m64jw2kk27fh7szp2a9-bzip2-1.0.8
  /nix/store/42xcbzrxkmsy26l6f2gywkyparavj6fd-xz-5.2.5
  /nix/store/4na42q4z37lz7w0nzn9xaapzn2xpdh88-openssl-1.1.1q
  /nix/store/6lp5n979d6dxddxasrm7rvpnmqkq8wf7-libcxxabi-11.1.0-dev
  /nix/store/72pq3pznvjfzd7inlb3g81v7xf9hp95a-gnu-config-2021-01-25
  /nix/store/78l73r385qhq0fcjh11xam2hcxjnhw2p-libxml2-2.9.14
  /nix/store/7976jz42z66bn128p8jh6v1xd3cvg5pd-hook
  /nix/store/7mbssjdl464l9gniknz8mp0wwiqrgz7n-libffi-3.4.2
  /nix/store/7wrwj968nhyp9ipwvbk5cyz20n62rcw0-llvm-11.1.0-lib
  /nix/store/8h3n392izqayw5dfimd4h24cfbvjv6n7-llvm-11.1.0
  /nix/store/8qr4a99758ck931xj312yzj032jlq8v3-libcxx-11.1.0
  /nix/store/ajz82lfc8i1rpqw123b7pvk6rg529ak2-gawk-5.1.1
  /nix/store/b3a0vk2pb05qh49ks0xzx8fxsy7c0r8k-gettext-0.21
  /nix/store/d1xp1ra0i1m55s0nsh7lv7ydigj10012-compiler-rt-libc-11.1.0-dev
  /nix/store/d6lp6lgn8qi871wnx6q2khhhsggzc3vs-gnugrep-3.7
  /nix/store/dc04skn4xjpam86fm9p7c5ra8wrawr30-compiler-rt-libc-11.1.0
  /nix/store/f41jl5i1sgx3xl8fwpxx1qp24f026ydy-xz-5.2.5-bin
  /nix/store/f8smfqh1462fq1n9lfl533pa5cj8a02b-coreutils-9.1
  /nix/store/fkz0lipg0ww1wd6dlp3f2dakzgfv7dam-gmp-with-cxx-6.2.1
  /nix/store/fmzlq75c1nrpd9hcihm4r26b7fxnncpq-DarwinTools-1
  /nix/store/h8ibd6ljb26xcany7c2any19b90inyl7-diffutils-3.8
  /nix/store/icd2j1qjmkvpgbh4g4x4jp8blya7whsl-cctools-binutils-darwin-949.0.1
  /nix/store/ifx9q1k4gh9p2zghypni9nnx51lcwz6x-sigtool-0.1.2
  /nix/store/iy6r8ddhk3i48jl1l1y8wsnasbb26x0n-expand-response-params
  /nix/store/j8w19vhk1yazy96zm22apalqm1jvvlrg-clang-wrapper-11.1.0
  /nix/store/jk4n55jw5hl0m9sp5yraw2rspvcmg80d-gzip-1.12
  /nix/store/kxmx58r7rwb0c63nj1g3fykzc3x1d42c-file-5.42
  /nix/store/m6qngxd9xkyv5ci1c686zmnhdjzqs25g-cctools-binutils-darwin-wrapper-949.0.1
  /nix/store/mqbxgphd9lsq1yd2q8lsx7x4qsjgyggk-nix-info
  /nix/store/n0bj0im39g6nagjav9l59hya8rbh2wl6-stdenv-darwin
  /nix/store/q5qjsnn2zwz6bjjpf6ia6v58lf1g5cq3-gnutar-1.34
  /nix/store/qyl9pbklsd1qx52jgr9zr645jj7j1g10-libiconv-50
  /nix/store/qzzicx3cjj00z0c4kx3v0919gqp40mnp-libSystem-11.0.0
  /nix/store/rfpj4283j0z9a7jhbg5nybhczai2b7xj-zlib-1.2.12
  /nix/store/ri7jbnrjvb2fcs1k1ihvygshx06zhalv-gnumake-4.3
  /nix/store/scc61slacjvms8nsc8scfa5sl9qwxjm9-bash-5.1-p16
  /nix/store/sgd84c5khradqb9sfzz38fkm7pzbxsxd-libcxx-11.1.0-dev
  /nix/store/sicqimzwjfkgxmf5vqga3srsnj2xdbl7-post-link-sign-hook
  /nix/store/sixpm4rrxhv7v7gxvm7w2z4y60m46774-gnused-4.8
  /nix/store/sksjxxs1b0rlbryxf6zqxcbksk905xyk-adv_cmds-119-locale
  /nix/store/vcgmzlzv9scgx1sjfbpb14k1qwsbbia2-signing-utils
  /nix/store/wy41fkahwg9krqg947f0dj9f1fkdv5wg-libtapi-1100.0.11
  /nix/store/xd8a5kqirpn561irafhhnjzp61917pxp-pcre-8.45
  /nix/store/xi0nfvicj7hn2lshdljbzzkni9abfmqn-ed-1.18
  /nix/store/xkmanz7k4c70qh0a6s70kl44c78p6sgi-findutils-4.9.0
  /nix/store/z0kl3bzmhwrvg9f1vb15m490kpz53ggc-clang-11.1.0-lib
  /nix/store/z70cb6whfk7bzvq6ky9xcx2l8lgr3z5m-cctools-port-949.0.1
  /nix/store/znp1a5sbwibkjiw7x5c01rclx1bwm74c-clang-11.1.0
  /nix/store/zp702lhqji4qzm7z79lnxcblgfp2javv-bzip2-1.0.8-bin
copying path '/nix/store/scc61slacjvms8nsc8scfa5sl9qwxjm9-bash-5.1-p16' from 'https://cache.nixos.org'...
copying path '/nix/store/fmzlq75c1nrpd9hcihm4r26b7fxnncpq-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/sksjxxs1b0rlbryxf6zqxcbksk905xyk-adv_cmds-119-locale' from 'https://cache.nixos.org'...
copying path '/nix/store/3azxqs52ssvb9m64jw2kk27fh7szp2a9-bzip2-1.0.8' from 'https://cache.nixos.org'...
copying path '/nix/store/dc04skn4xjpam86fm9p7c5ra8wrawr30-compiler-rt-libc-11.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/zp702lhqji4qzm7z79lnxcblgfp2javv-bzip2-1.0.8-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/d1xp1ra0i1m55s0nsh7lv7ydigj10012-compiler-rt-libc-11.1.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/xi0nfvicj7hn2lshdljbzzkni9abfmqn-ed-1.18' from 'https://cache.nixos.org'...
copying path '/nix/store/iy6r8ddhk3i48jl1l1y8wsnasbb26x0n-expand-response-params' from 'https://cache.nixos.org'...
copying path '/nix/store/ajz82lfc8i1rpqw123b7pvk6rg529ak2-gawk-5.1.1' from 'https://cache.nixos.org'...
copying path '/nix/store/72pq3pznvjfzd7inlb3g81v7xf9hp95a-gnu-config-2021-01-25' from 'https://cache.nixos.org'...
copying path '/nix/store/ri7jbnrjvb2fcs1k1ihvygshx06zhalv-gnumake-4.3' from 'https://cache.nixos.org'...
copying path '/nix/store/sixpm4rrxhv7v7gxvm7w2z4y60m46774-gnused-4.8' from 'https://cache.nixos.org'...
copying path '/nix/store/jk4n55jw5hl0m9sp5yraw2rspvcmg80d-gzip-1.12' from 'https://cache.nixos.org'...
copying path '/nix/store/7976jz42z66bn128p8jh6v1xd3cvg5pd-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/qzzicx3cjj00z0c4kx3v0919gqp40mnp-libSystem-11.0.0' from 'https://cache.nixos.org'...
copying path '/nix/store/21vk5rmy1a4p3kx22wzrdrllcmy8x5hl-libcxxabi-11.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/7mbssjdl464l9gniknz8mp0wwiqrgz7n-libffi-3.4.2' from 'https://cache.nixos.org'...
copying path '/nix/store/8qr4a99758ck931xj312yzj032jlq8v3-libcxx-11.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/6lp5n979d6dxddxasrm7rvpnmqkq8wf7-libcxxabi-11.1.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/fkz0lipg0ww1wd6dlp3f2dakzgfv7dam-gmp-with-cxx-6.2.1' from 'https://cache.nixos.org'...
copying path '/nix/store/sgd84c5khradqb9sfzz38fkm7pzbxsxd-libcxx-11.1.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/f8smfqh1462fq1n9lfl533pa5cj8a02b-coreutils-9.1' from 'https://cache.nixos.org'...
copying path '/nix/store/qyl9pbklsd1qx52jgr9zr645jj7j1g10-libiconv-50' from 'https://cache.nixos.org'...
copying path '/nix/store/h8ibd6ljb26xcany7c2any19b90inyl7-diffutils-3.8' from 'https://cache.nixos.org'...
copying path '/nix/store/xkmanz7k4c70qh0a6s70kl44c78p6sgi-findutils-4.9.0' from 'https://cache.nixos.org'...
copying path '/nix/store/b3a0vk2pb05qh49ks0xzx8fxsy7c0r8k-gettext-0.21' from 'https://cache.nixos.org'...
copying path '/nix/store/wy41fkahwg9krqg947f0dj9f1fkdv5wg-libtapi-1100.0.11' from 'https://cache.nixos.org'...
copying path '/nix/store/q5qjsnn2zwz6bjjpf6ia6v58lf1g5cq3-gnutar-1.34' from 'https://cache.nixos.org'...
copying path '/nix/store/z70cb6whfk7bzvq6ky9xcx2l8lgr3z5m-cctools-port-949.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/4na42q4z37lz7w0nzn9xaapzn2xpdh88-openssl-1.1.1q' from 'https://cache.nixos.org'...
copying path '/nix/store/2qv84lnl9m6spbfvqdp0a5zc20zg06nd-patch-2.7.6' from 'https://cache.nixos.org'...
copying path '/nix/store/xd8a5kqirpn561irafhhnjzp61917pxp-pcre-8.45' from 'https://cache.nixos.org'...
copying path '/nix/store/ifx9q1k4gh9p2zghypni9nnx51lcwz6x-sigtool-0.1.2' from 'https://cache.nixos.org'...
copying path '/nix/store/d6lp6lgn8qi871wnx6q2khhhsggzc3vs-gnugrep-3.7' from 'https://cache.nixos.org'...
copying path '/nix/store/sicqimzwjfkgxmf5vqga3srsnj2xdbl7-post-link-sign-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/mqbxgphd9lsq1yd2q8lsx7x4qsjgyggk-nix-info' from 'https://cache.nixos.org'...
copying path '/nix/store/vcgmzlzv9scgx1sjfbpb14k1qwsbbia2-signing-utils' from 'https://cache.nixos.org'...
copying path '/nix/store/42xcbzrxkmsy26l6f2gywkyparavj6fd-xz-5.2.5' from 'https://cache.nixos.org'...
copying path '/nix/store/rfpj4283j0z9a7jhbg5nybhczai2b7xj-zlib-1.2.12' from 'https://cache.nixos.org'...
copying path '/nix/store/f41jl5i1sgx3xl8fwpxx1qp24f026ydy-xz-5.2.5-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/1m5r3xkfbgcnfrz99kkjdb5h29v36rp6-binutils-2.38' from 'https://cache.nixos.org'...
copying path '/nix/store/kxmx58r7rwb0c63nj1g3fykzc3x1d42c-file-5.42' from 'https://cache.nixos.org'...
copying path '/nix/store/78l73r385qhq0fcjh11xam2hcxjnhw2p-libxml2-2.9.14' from 'https://cache.nixos.org'...
copying path '/nix/store/7wrwj968nhyp9ipwvbk5cyz20n62rcw0-llvm-11.1.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/z0kl3bzmhwrvg9f1vb15m490kpz53ggc-clang-11.1.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/8h3n392izqayw5dfimd4h24cfbvjv6n7-llvm-11.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/znp1a5sbwibkjiw7x5c01rclx1bwm74c-clang-11.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/icd2j1qjmkvpgbh4g4x4jp8blya7whsl-cctools-binutils-darwin-949.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/m6qngxd9xkyv5ci1c686zmnhdjzqs25g-cctools-binutils-darwin-wrapper-949.0.1' from 'https://cache.nixos.org'...
copying path '/nix/store/j8w19vhk1yazy96zm22apalqm1jvvlrg-clang-wrapper-11.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/n0bj0im39g6nagjav9l59hya8rbh2wl6-stdenv-darwin' from 'https://cache.nixos.org'...
Starting a devbox shell...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.5.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by ebauger** [comment]
For the context: I use the wrapper layer of Nix Package Manager and docker with https://www.jetpack.io/blog/devbox/

**Comment by drupol** [comment]
The `imap` extension is a core PHP extension and therefore, it is already available.

On my side, I succeeded to test this by doing:

1. Create a random `composer.json` file containing: `"ext-imap": "*"` in `require` or `require-dev`.
2. Run: `nix shell github:loophp/nix-shell#php81 --impure` in the same directory
3. `php -m`

**Comment by drupol** [comment]
Dear @ebauger ,

I guess we can now close this issue ?

**Comment by norberttech** [comment]
@drupol any idea why [imap is not enable at darwin](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/php/8.1.nix#L51)?

**Comment by drupol** [comment]
No, but this is a good question, I guess @etu or @jtojnar might help here.

**Comment by etu** [comment]
So I did some digging into this, because it's a very good question.

The reason for it not being enabled by default on darwin is because it wasn't enabled by default *before* we migrated to our modular build of PHP.

The latest release branch *before* the modular build was 19.09 and here's the PHP definition from that time, where I have highlighted the line that we emulated by leaving it disabled after that big refactoring: https://github.com/NixOS/nixpkgs/blob/nixos-19.09/pkgs/development/interpreters/php/default.nix#L18 -- so in 19.09 it was disabled on darwin.

Inspecting that line further, the last time it was changed before that I linked it as of 19.09 was as part of a refactoring in July 2018: https://github.com/NixOS/nixpkgs/commit/ae9932e4312991cc3d3d611f541bab7d6567f392 -- however, this commit actually didn't change the fact if it was enabled or not.

So looking back ever further I found this, the commit `php: Now builds on darwin` authored in June 2016: https://github.com/NixOS/nixpkgs/commit/d4babc49dcf0db770cc49f6e1cda7b89fdf0fca7

That made a bunch of changes to the PHP definition, in which, one of the changes was to disable the imap extension on darwin.

So I would assume that at that time, in June 2016, either `uwimap` as a dependency or the `imap` PHP extension itself actually didn't build on darwin, hence it was disabled to make PHP build at all on darwin.

Of course... now... 6 years later... this may not be a problem anymore and maybe we should revisit this :wink:
